cns/service/main.go

```diff
@@ -1252,3 +1253,108 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 
 	return nil
 }
+
+func SetMellanoxRegKey() {
+	var mellanoxSearchString = "*Mellanox*"
+	var priorityVLANTagIdentifier = "*PriorityVLANTag"
+	var registryKeyPrefix = "HKLM:\\System\\CurrentControlSet\\Control\\Class\\"
+
+	//get adapter name
+	cmd := fmt.Sprintf("Get-NetAdapter | Where-Object { $_.InterfaceDescription -like \"%s\" } | Select-Object -ExpandProperty Name", mellanoxSearchString)
+	adapatername, err := platform.ExecutePowershellCommand(cmd)
+
+	if err != nil {
+		logger.Printf("Error while executing pwsl command to get net adapter list: %v", err)
+		return
+	}
+	if adapatername == "" {
+		logger.Printf("No Network adapter found with \"%s\" in description", mellanoxSearchString)
+		return
+	}
+
+	logger.Printf("Name of Mellanox adapter : %v", adapatername)
+
+	//Find if adapter with Priority vlan tag (version 4 or up) or not(version 3)
+	cmd = fmt.Sprintf("Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty Name", priorityVLANTagIdentifier, adapatername)
+	adapterNamewithVLANTag, err := platform.ExecutePowershellCommand(cmd)
+	if err != nil {
+		logger.Printf("Error while executing pwsh command to get VLAN Tag advance property of %s: %v", adapatername, err)
+		return
+	}
+	if adapterNamewithVLANTag != "" {
+		logger.Printf("Found %s in adapter's advanced properties", priorityVLANTagIdentifier)
+
+		//set reg value for Priorityvlantag of adapater to 3
+		cmd = fmt.Sprintf("Set-NetAdapterAdvancedProperty -Name \"%s\" -RegistryKeyword \"%s\" -RegistryValue 3", adapatername, priorityVLANTagIdentifier)
+		_, err := platform.ExecutePowershellCommand(cmd)
+		if err != nil {
+			logger.Printf("Error while setting up registery value for PriorityVLANTag for adapater: %v", err)
+		}
+		//check if value is set or not
+		cmd = fmt.Sprintf(" Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty RegistryValue", priorityVLANTagIdentifier, adapatername)
+		regvalue, err := platform.ExecutePowershellCommand(cmd)
+		if err != nil {
+			logger.Printf("Error while checking registery value for PriorityVLANTag for adapater: %v", err)
+		}
+		if regvalue != "3" {
+			logger.Printf("Failure while setting Mellanox Network Adapter: \"%s\" with \"%s\" property value as 3", adapatername, priorityVLANTagIdentifier)
+		} else {
+			logger.Printf("Successfully set Mellanox Network Adapter: \"%s\" with \"%s\" property value as 3", adapatername, priorityVLANTagIdentifier)
+		}
+	} else {
+		//looks like its version 3 so vlantag not available so need to follow other route
+		logger.Printf("Could not find \"%s\" in adapter's advanced properties", priorityVLANTagIdentifier)
+		logger.Printf("Proceeding in a different way")
```

Comment:
having 'Proceeding in a different way' in telemetry doesn't help anyone.

Reply:
cleaned up unnecessary comments